reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/job/parameter/JobParameter.java

```diff
@@ -0,0 +1,136 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.job.parameter;
+
+import java.io.IOException;
+import java.time.Instant;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.ScheduledJobParameter;
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.Schedule;
+
+/**
+ * A report scheduled job parameter.
+ *
+ * <p>It adds an additional "reportDefinition" field to {@link ScheduledJobParameter}, which stores
+ * the index the job runner will watch.
+ */
+public class JobParameter implements ScheduledJobParameter {
+
+  private final String jobName;
+  private final Instant enabledTime;
+  private final String reportDefinitionId;
+  private final boolean isEnabled;
+  private final Schedule schedule;
+  private final Instant lastUpdateTime;
+  private Long lockDurationSeconds;
+
+  public JobParameter(
+      String jobName,
+      Instant enabledTime,
+      String reportDefinitionId,
+      boolean isEnabled,
+      Schedule schedule,
+      Instant lastUpdateTime) {
+    this.jobName = jobName;
+    this.enabledTime = enabledTime;
+    this.reportDefinitionId = reportDefinitionId;
+    this.isEnabled = isEnabled;
+    this.schedule = schedule;
+    this.lastUpdateTime = lastUpdateTime;
+  }
+
+  @Override
+  public String getName() {
+    return this.jobName;
+  }
+
+  @Override
+  public Instant getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  @Override
+  public Instant getEnabledTime() {
+    return this.enabledTime;
+  }
+
+  @Override
+  public Schedule getSchedule() {
+    return this.schedule;
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return this.isEnabled;
+  }
+
+  @Override
+  public Long getLockDurationSeconds() {
+    return this.lockDurationSeconds;
+  }
+
+  public String getReportDefinitionId() {
+    return this.reportDefinitionId;
+  }
+
+  //  public void setJobName(String jobName) {
+  //    this.jobName = jobName;
+  //  }
+  //
+  //  public void setLastUpdateTime(Instant lastUpdateTime) {
+  //    this.lastUpdateTime = lastUpdateTime;
+  //  }
+  //
+  //  public void setEnabledTime(Instant enabledTime) {
+  //    this.enabledTime = enabledTime;
+  //  }
+  //
+  //  public void setEnabled(boolean enabled) {
+  //    isEnabled = enabled;
+  //  }
+  //
+  //  public void setSchedule(Schedule schedule) {
+  //    this.schedule = schedule;
+  //  }
+  //
+  //  public void setReportDefinitionId(String reportDefinitionId) {
+  //    this.reportDefinitionId = reportDefinitionId;
+  //  }
+
+  public void setLockDurationSeconds(Long lockDurationSeconds) {
```

Comment:
Why is this class mutable?

Reply:
Because it doesn't need locking when we created the scheduled job, but it needs locking when we push the job to queue. That's why I left this variable with a setter. But I can add locking for scheduled job too, I just feel like there is no need for that. 

In job-scheduler document it says `ScheduledJob executions are not a singleton runner and if a job execution takes longer than the interval/cron time then there can be two or more execution running at a time.`

In our use case, the "job execution" is nothing complex, just pushing scheduled job to queue index, it's just a "save to index" action. And it won't take longer than the interval time we set.

Reply:
With that information, the appropriate object can be created with proper value during construction to avoid mutability.

Reply:
will make it final and use constructors for initialization, like other variables